erpnext/accounts/report/general_ledger/general_ledger.js

```diff
@@ -82,11 +85,12 @@ frappe.query_reports["General Ledger"] = {
 					frappe.query_report_filters_by_name.party_name.set_value("");
 					return;
 				}
-
-				var fieldname = frappe.scrub(party_type) + "_name";
-				frappe.db.get_value(party_type, party, fieldname, function(value) {
-					frappe.query_report_filters_by_name.party_name.set_value(value[fieldname]);
-				});
+				if (party_type != "Student" && party_type != "Shareholder") {
```

Comment:
such type of hardcoding should be avoided. Let there be a global object that will return the `_name` property of the given party.